### ISSUE TITLE
Bug in func MergeInto of TimestampCache

### ISSUE THREAD

**Issue by changrui82** [cockroachdb/cockroach#3554]
```
func (tc *TimestampCache) MergeInto(dest *TimestampCache, clear bool) {
	if clear {
		dest.cache.Clear()
		dest.lowWater = tc.lowWater
		dest.latest = tc.latest
	} else {
		if dest.lowWater.Less(tc.lowWater) {
			dest.lowWater = tc.lowWater
		}
		if dest.latest.Less(tc.latest) {
			tc.latest = dest.latest
		}
	}
	tc.cache.Do(func(k, v interface{}) {
		dest.cache.Add(k, v)
	})
}
```
I thinks `tc.latest = dest.latest` should be `dest.latest = tc.latest`, right?

**Comment by tschottdorf** [comment]
Yes. I think these `if x.Less(y) { x = y }` can be written more concisely as `x.Forward(y)`. Wanna open a PR?

**Comment by AbhishekSaha** [comment]
Hi, I'm not OP but I tried fixing the bug. I submitted the Pull Request here: https://github.com/cockroachdb/cockroach/pull/3797

I'm new to contributing to Open Source projects and Cockroach, but this project seems pretty cool and I'd like to help. I'm not sure where the existing tests @tschottdorf mentions exist in the project, would you mind telling me where I could find their path?

**Comment by tschottdorf** [comment]
Hey @AbhishekSaha, I've commented on your PR #3797.

**Comment by tamird** [comment]
Fixed in #3797.
